File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        // Create output list
        List<List<String>> output = new ArrayList<>();
        // Create a map: key = sorted element, value = list of indexes
        Map<String, List<Integer>> dic = new HashMap<>();
        // Iterate through input
        for (int i = 0; i < strs.length; i++) {
            char[] arr = strs[i].toCharArray();
            Arrays.sort(arr);
            String sort = new String(arr);
            dic.putIfAbsent(sort, new ArrayList<>());
            dic.get(sort).add(i);
        }
        // Iterate through map values
        for (List<Integer> indexes: dic.values()) {
            List<String> ana = new ArrayList<>();
            // Iterate through list
            for (Integer index : indexes) {
            ana.add(strs[index]);
            }
            // Append list to output
            output.add(ana);
        }
        // Return output
        return output;
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        # Create output list
        output = []
        # Create a map: key = sorted element, value = list of indexes
        dic = {}
        # Iterate through input
        for i in range(len(strs)):
            # If sorted element not in map,
            sort = ''.join(sorted(strs[i]))
            if sort not in dic:
                # Add sorted element and a list with the index
                dic[sort] = [i]
            # Else,
            else:
                # Append index to list of sorted element
                dic[sort].append(i)
        # Iterate through map values
        for i_list in dic.values():
            # Iterate through list
            for i in range(len(i_list)):
                # Change each index to it's element
                i_list[i] = strs[i_list[i]]
            # Append list to output
            output.append(i_list)
        # Return output
        return output